File: asgn2/python_code/task3_java/Attendant.java
package assg2task3;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author kenchan
 */
public class Attendant {
    protected String name;
    protected double weight;
    protected double wealth;
    protected int position;
    public Attendant(String Name, double Weight, double Wealth, int Position){
        this.name = Name;
        this.weight = Weight;
        this.wealth = Wealth;
        this.position = Position;
    }
    public double MoveTo(Room Destination){
        int distance = Math.abs(this.position - Destination.GetPosition());
        double calories_consumed = distance * this.weight;
        this.position = Destination.GetPosition();
        System.out.println(this.name + " moved to " + Destination.GetName() + "." );
        return calories_consumed;
    }
    public double MoveTo(Fixture Destination){
        int distance = Math.abs(this.position - Destination.GetPosition());
        double calories_consumed = distance * this.weight;
        this.position = Destination.GetPosition();
        System.out.println(this.name + " moved to " + Destination.GetName() + "." );
        return calories_consumed;
    }
    public double MoveTo(Room Destination, Fixture Obj){
        int distance = Math.abs(Obj.GetPosition() - Destination.GetPosition());
        double calories_consumed = distance * Obj.GetWeight();
        Obj.position = Destination.GetPosition();
        System.out.println(Obj.name + " was moved to " + Destination.GetName() + "." );
        return calories_consumed;
    }
    public void CarryTo(Fixture Obj, Room Destination){
        double calories_consumed = 0;
        //Move the attendant to the object first
        calories_consumed += this.MoveTo(Obj);
        //Move the attendant to the destination
        calories_consumed += this.MoveTo(Destination);
        //Move the object to the destination 
        calories_consumed += this.MoveTo(Destination, Obj);
        //Print calories consumed
        System.out.println(this.name + " consumed " + calories_consumed + " calories." );
    }
    public String GetName(){
        return this.name;
    }
    public int GetPosition(){
        return this.position;
    }
    public double GetWeight(){
        return this.weight;
    }
    public double Pay(double Amount){
        if (Amount <= this.wealth){
            this.wealth -= Amount;
            System.out.println(this.name + " paid " + Amount + " dollars.");
            return Amount;
        }
        else{
            return -1;
        }
    }
}

File: Programming_Languages/asgn2/python_code/task3_python/Attendant.py
#this is the python Attendent!

import Attendant
import DumbFixture
import Room
import SmartFixture
import abc

class Attendant:
    __metaclass__=abc.ABCMeta
    def __init__(self,Name,Weight,Wealth,Position):
        self._name=Name
        self._weight=Weight
        self._wealth=Wealth
        self._position=Position

    def MoveTo(self,Destination):
        distance= abs(self._position - Destination.GetPosition())
        calories_consumed = distance * self._weight
        self._position=Destination.GetPosition()
        print ('%s moved to %s.' % self._name, Destination.GetName())
        return calories_consumed

    def MoveTo(self,Destination, Obj):
        distance= abs(Obj.GetPosition() - Destination.GetPosition())
        calories_consumed = distance * Obj.GetWeight()
        Obj.position=Destination.GetPosition()
        print ('%s moved to %s.' % Obj.GetName, Destination.GetName())
        return calories_consumed

    def CarryTo(self,Obj, Destination):
        calories_consumed=0.0
        calories_consumed= calories_consumed+self.MoveTo(Obj)
        calories_consumed= calories_consumed+self.MoveTo(Destination)
        calories_consumed= calories_consumed+self.MoveTo(Destination, Obj)
        print ('%s consumed %.2f calories.' % self._name, calories_consumed)

    def GetName(self):
        return self._name

    def GetPosition(self):
        return self._position

    def GetWeight(self):
        return self._weight

    def Pay(self,Amount):
        if (Amount<=self._wealth):
            self._wealth = self._wealth - Amount
            print ('%s paid %.2f dollars.' % self._name,Amount)
            return Amount
        else:
            return -1
